src/omnicore/omnicore.cpp

```diff
@@ -2367,15 +2372,38 @@ bool mastercore::UseEncodingClassC(size_t nDataSize)
     return nTotalSize <= nMaxDatacarrierBytes && fDataEnabled;
 }
 
+/**
+ * Determines whether Class D is active & whether the compressed payload is suitable size.
+ *
+ * @param nDataSize The length of the payload
+ * @return True, if Class D is enabled and the payload is small enough
+ */
+bool mastercore::UseEncodingClassD(size_t nDataSize)
+{
+    size_t nTotalSize = nDataSize + GetOmMarker(OMNI_CLASS_D).size(); // Marker "ol"
+    int nBlockNow = GetHeight();
+    bool fDataEnabled = GetBoolArg("-datacarrier", true);
+    bool fClassDEnabled = IsFeatureActivated(FEATURE_CLASS_D, nBlockNow);
+    if (nDataSize == 0) { // no compressed payload supplied
```

Comment:
Aah, I see!

> I could move the check for an empty compressed payload into WalletTxBuilder() itself rather than UseEncodingClassD() if you think that is a more suitable place for it?

Sounds good to me, I think. :)